### PR TITLE
Add simplex action space and dirichlet action distribution

### PR DESCRIPTION
**Pull request by zekun-shi**
## What do these changes do?

Add a custom action space (Simplex) that inherit gym space class, which can be used to model action that represents resource allocation. Also, the accompanying action distribution (Dirichlet) is added.

### CODE REVIEW

**Review by ericl — COMMENTED** [review]
Looks good (I pushed some style changes). Does it also make sense to add an entry to https://github.com/ray-project/ray/blob/master/python/ray/rllib/test/test_supported_spaces.py#L17 ?

File: python/ray/rllib/models/extra_spaces.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    def seed(self, seed):
+        self.np_random.seed(seed)
+
+    def sample(self):
+        return np.random.dirichlet(
+            self.concentration, size=self.shape[:-1]).astype(self.dtype)
+
+    def contains(self, x):
+        assert x.shape == self.shape, (x.shape, self.shape)
+        assert np.allclose(np.sum(x, axis=-1), np.ones_like(x[..., 0]))
```
Comment (ericl) [inline python/ray/rllib/models/extra_spaces.py]: Does it make sense to return False instead of asserting here?

Comment (Szkered) [reply python/ray/rllib/models/extra_spaces.py]: Yes it should actual return False when the sample is not in the given space. I've updated this part.

**Review by ericl — APPROVED** [review]

**Comment by ericl** [comment]
Merged, thanks!

**Pull request merged**